### PR TITLE
Debt -  5067 - Migrate `admin` to `apps/web`

### LINKED ISSUE #5067: Move in Admin
# This is the ~~second~~ fourth step in the Combine frontend epic.

> **Warning**
> This is a disruptive process.  Notify your team that you will be working on this workspace.

## :dart: Goals
1. Move the admin project to the new structure
2. Move as quickly as possible to minimize disruptions
3. Leave the tidying and refactoring for later

## :heavy_check_mark:  Acceptance Criteria
- [ ] Entire app continues to function as before
- [ ] Admin now lives in the new directory structure

## :notebook:  References
- Slack threads: [brainstorming](https://talent-cloud.slack.com/archives/CB424V61J/p1670440607411319) and [discussion](https://talent-cloud.slack.com/archives/CB424V61J/p1670610552790639)
- Items of concern during the moves:
  - Updating nginx configs, config.ejs
  - Maintenance scripts
  - translation files and workflows
  - feature flags, index.html, and webpack
  - jest, cypress
  - hydrogen settings

### PR DESCRIPTION
**Pull request by esizer**
🤖 Resolves #5067

## 👋 Introduction

This moves the `frontend/admin` workspace to `app/web`

## 🧪 Testing

Assist reviewers with steps they can take to test that the PR does what it says it does.

1. Rebuild the docker containers
    - `docker-compose down`
    - `docker-compose up --build --detach`
3. Build the app `npm run build`
4. Test all pages on admin
5. Review code for references to admin and confirm they have all been removed

### CODE REVIEW

File: .bundlewatch.config.js
Comment (mnigh) [inline .bundlewatch.config.js:14]: Bump bundle maxSize [because](https://service.bundlewatch.io/results?d=details%7CrepoOwner%7CGCTC-NTGC%7CrepoName%7Cgc-digital-talent%7CrepoCurrentBranch%7Cdebt_5067_migrate-admin%7CrepoBranchBase%7Cmain%7CcommitSha%7C0ee0429ecee8d03d4b6c43bca37a03c0d95170d4%7Cresults%7Cstatus%7Cfail%7CfullResults%7CfilePath%7Capps/web/dist/app.cb87bde8ab900ba5d165.js%7Csize%7CbaseBranchSize%7CmaxSize%7Cfrontend/admin/dist/app.1ff83945327e804d1e35.js%7Cremoved%7Capps/web/dist/app.687842b14e934d4e8738.js%7Csummary%7CmaxSize%2Bcheck%2Bfailed%5E5ZZH%7C0%7C5HJ4%7C0%7C53AO%7C0%7C0%7C5FMK%7C0%5E%5E$0%7C$1%7C2%7C3%7C4%7C5%7C6%7C7%7C8%7C9%7CA%5D%7CB%7C$C%7CD%7CE%7C@$F%7CG%7CC%7CD%7CH%7CP%7CI%7CQ%7CJ%7CR%5D%7C$F%7CK%7CC%7CL%7CH%7CS%7CI%7CT%7CJ%7CU%5D%7C$F%7CM%7CC%7CL%7CH%7CV%7CI%7CW%7CJ%7CX%5D%5D%7CN%7CO%5D%5D) of merge
```suggestion
      maxSize: "300 kB",
```

File: maintenance/scripts/setup.sh
Comment (esizer) [inline maintenance/scripts/setup.sh:19]: @petertgiles This still seems to create a directory (`.env/.env.example`) instead of copying the file :/.

Looks like it is also happening for the `api/.env.example` as well.

**Review by petertgiles — CHANGES_REQUESTED** [review]
Looking great!  A few things to check on, then I'll push it out to the dev server.

File: apps/web/.storybook/main.js
Comment (petertgiles) [reply apps/web/.storybook/main.js:3]: Happening in Chromatic, too.

File: maintenance/scripts/setup.sh
Comment (esizer) [reply maintenance/scripts/setup.sh:19]: okay, so I just moved the`.env` back to `apps/web` so we have one file now. Seems to be working so far... 😅 

[move .env back to apps/web](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5528/commits/5695b38c0bdac0b92c37c4a3e7cb004bf7bd4e7c)

File: apps/web/.storybook/main.js
Comment (esizer) [reply apps/web/.storybook/main.js]: Removed in https://github.com/GCTC-NTGC/gc-digital-talent/pull/5528/commits/56282deeeb00e41e3ae7446e13506a78f0b27bc0

File: apps/web/src/lang/fr.json
Comment (esizer) [reply apps/web/src/lang/fr.json:1428]: I can't wait until this is part of our CI in someway :P

Removed in [remove orphaned strings](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5528/commits/f5415abaa7f66d6f2684a4c0b8f2a27cc89aa163)

File: apps/web/.storybook/main.js
Comment (esizer) [reply apps/web/.storybook/main.js:3]: Oops, looks like I missed an important css file!

Added in [add missing css file to storybook](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5528/commits/a85a6f5fae26bb569802bf940ffb8d4597f4cc87)

**Comment by vd1992** [comment]
some additional stories broke
https://61e099a1bb1465003a73d3ce-msjzcoxdcw.chromatic.com/?path=/story/forms-search-form--default
https://61e099a1bb1465003a73d3ce-msjzcoxdcw.chromatic.com/?path=/story/pages-search-page--search-page

prior to this a story was already broken #5431

**Comment by esizer** [comment]
> some additional stories broke

Weird, those stories should have never been working because they were missing a required prop :/ Fixed in [fix search page stories](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5528/commits/8078a3780898a1c47d91cfe9acc4b4a473acb7a0)

> I noticed the story count went way down, appears all the /common stories are out?

Oops, that was my fault. I commented the line out to speed it up while testing. Added back in [add common stories back](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5528/commits/7680f533f1b3ea7ccc4ac0adb758184765771c2c)

File: maintenance/scripts/setup.sh
Comment (petertgiles) [reply maintenance/scripts/setup.sh:19]: I just did a fresh setup after your changes and it seems to work great now. :+1:

File: apps/web/.storybook/main.js
Comment (petertgiles) [reply apps/web/.storybook/main.js:3]: Uh oh, the admin stuff is just gone now in storybook.
image

**Review by petertgiles — COMMENTED** [review]
Looking good, just storybook and maybe source maps left to check on.

File: apps/web/.storybook/main.js
Comment (esizer) [reply apps/web/.storybook/main.js:3]: Is this local or in chromatic?

File: infrastructure/conf/nginx-conf-deploy/default
Comment (petertgiles) [reply infrastructure/conf/nginx-conf-deploy/default:96]: I think both.  I'm pretty sure we generate source maps in production and we don't have a reason to obfuscate the source, right?

File: apps/web/.storybook/main.js
Comment (petertgiles) [reply apps/web/.storybook/main.js:3]: Sorry, yeah, this problem was local.

Comment (esizer) [reply apps/web/.storybook/main.js:3]: Hmm, i am seeing it. 🤔 

Screenshot 2023-01-30 at 3 16 23 PM

File: infrastructure/conf/nginx-conf-deploy/default
Comment (esizer) [reply infrastructure/conf/nginx-conf-deploy/default:96]: Makes sense, add in [add map files to static file routing](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5528/commits/572a1d0e8c55d5542b08b7d43583e73d6f47dc96)

File: apps/web/.storybook/main.js
Comment (petertgiles) [reply apps/web/.storybook/main.js:3]: Sorry, my bad.  I think it launched storybook from the common directory accidentally. :face_with_spiral_eyes: I tried again and it looks great.

File: infrastructure/conf/nginx-conf-deploy/default
Comment (petertgiles) [reply infrastructure/conf/nginx-conf-deploy/default:96]: Looks great!  I'm just building in Azure now.

Comment (esizer) [reply infrastructure/conf/nginx-conf-deploy/default:96]: 🤞

Comment (petertgiles) [reply infrastructure/conf/nginx-conf-deploy/default:96]: Everything looks good!

**Review by mnigh — CHANGES_REQUESTED** [review]
Amazing work! Some minor 🧹 and then some things that are/might have been existing before this PR. Feel free to spin those off into separate issues if you think they will take too long.

Screen Shot 2023-01-31 at 09 44 13
 
Screen Shot 2023-01-31 at 09 44 36

Screen Shot 2023-01-31 at 09 45 50

Screen Shot 2023-01-31 at 09 48 55

Screen Shot 2023-01-31 at 09 27 44
Screen Shot 2023-01-31 at 09 31 23
Screen Shot 2023-01-31 at 09 35 14

Screen Shot 2023-01-31 at 09 22 07

Screen Shot 2023-01-31 at 09 22 56

Screen Shot 2023-01-31 at 09 24 18

**Comment by esizer** [comment]
@mnigh I think most of these are exiting issues unrelated to this PR so I can work to create tickets for each. For some, I'm not sure:

I have no idea what that `.gitattributes` file does so I left in untouched. @petertgiles Do you know what this does?

`launch.json.example`, again, I don't know that this file is doing 😢 

`e2e/README` these are valid references since we have folders in there that separates the spec files.

For the pool candidates story, this will be resolved in #5068

**Comment by petertgiles** [comment]
> I have no idea what that `.gitattributes` file does so I left in untouched. @petertgiles Do you know what this does?
> 
I've used this file to force Git to use LF line endings even on Windows.  I think it's a good idea to ensure that this continues going forward.  I'm not sure about the Linguist reference.  Maybe a documentation tool being used with Hydrogen?  @substrae ?

> `launch.json.example`, again, I don't know that this file is doing cry
>
These are debugging profiles for VS Code.  I didn't want to make eight debugging profiles for each combination of browser of subproject so I just made an example file.  Once _common_ is merged in it might be worth making active debugging profiles again!  I'd suggest leaving these alone for now.  They're probably back from the Laravel Mix days and won't work right now anyway.  :laughing:

**Comment by esizer** [comment]
Okay, so do we think it is safe to leave those references alone for now and maybe revisit them after all the merging is complete? @mnigh + @petertgiles

**Review by mnigh — APPROVED** [review]

**Pull request merged**
